packages/amplify_auth_cognito/test/amplify_auth_cognito_confirmSignIn_test.dart

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_auth_cognito/method_channel_auth_cognito.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+void main() {
+  const MethodChannel authChannel = MethodChannel('com.amazonaws.amplify/auth_cognito');
+  const MethodChannel coreChannel = MethodChannel('com.amazonaws.amplify/core');
+
+  Amplify amplify = new Amplify();
+  AmplifyAuthCognito auth = AmplifyAuthCognito();
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  int testCode = 0;
+
+  setUp(() {
+    authChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "confirmSignIn") {
+        assert(methodCall.arguments["data"] is Map);
+        assert(methodCall.arguments["data"]["confirmationCode"] is String);
+        switch(testCode) {
+          case 0:
+            return {
+              "isSignedIn": false,
+              "nextStep": {
+                "signInStep": "DONE",
+                "codeDeliveryDetails":  { "atttibuteName": "email" }
+              }
+            };
+          case 1: 
+            return throw PlatformException(code: "AMPLIFY_EXCEPTION", message: "AMPLIFY_CONFIRM_SIGNIN_FAILED", details: {} );
+        }
+      } else {
+        return true;
+      }     
+    });
+    coreChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      return true;
+    });
+  });
+
+  tearDown(() {
+    authChannel.setMockMethodCallHandler(null);
+    coreChannel.setMockMethodCallHandler(null);
+  });
+
+  test('confirmSignIn request returns a SignInResult', () async {
+    await amplify.addPlugin(authPlugins: [auth]);
+    await amplify.configure("{}");
+    var res = await Amplify.Auth.confirmSignIn(confirmationValue: "iAmLegit");
+    expect(res, isInstanceOf<SignInResult>());
+    expect(res.isSignedIn, false);
+    expect(res.nextStep, isInstanceOf<AuthNextSignInStep>());
```

Comment:
Can we instead assert against a full response object that we expect.

Reply:
Yes - although I still check the values individually.